presto-main/src/main/java/io/prestosql/sql/planner/PlanFragmenter.java

```diff
@@ -82,17 +85,22 @@
  */
 public class PlanFragmenter
 {
+    private static final String TOO_MANY_STAGES_MESSAGE = "If the query contains multiple DISTINCTs, please set the 'use_mark_distinct' session property to false. " +
```

Comment:
This message is not very precise. Regarding DISTINCT, it's only if there are multiple aggregates in the same SELECT statement with DISTINCT over different columns. Also, CTE is a term of art that may not be understood by most people -- it's not even something the spec references. I'd say "WITH clauses".

Reply:
Note that this message is not being introduced here. See https://github.com/prestosql/presto/blob/7c346e723b343dbcf9dae99ef4687b42d2aec2ae/presto-main/src/main/java/io/prestosql/sql/planner/PlanFragmenter.java#L123-L125
Of course, we can improve it (later; or in this PR).

Reply:
I improved the message a bit (see fixup). I don't know how to phrase the part about DISTINCT within same SELECT statement.

Reply:
How about:

"If the query contains multiple aggregates with DISTINCT over different columns, please set [...]. If the query contains WITH clauses that are referenced more than once, please create temporary table(s) for the queries in those clauses."

Reply:
applied